configuration/src/lib.rs

```diff
@@ -336,8 +336,19 @@ impl NomadConfig {
     }
 
     /// Get a reference to the nomad config's gas map.
-    pub fn gas(&self) -> &HashMap<String, NomadGasConfig> {
-        &self.gas
+    pub fn gas(&self) -> HashMap<String, NomadGasConfig> {
```

Comment:
This will clone the entire contents, which i think is not desirable

Reply:
Can't return `&HashMap` since return type and actual held type are not same (`NomadGasConfig` vs `NomadGasConfig[s]`). But we can return ref to underlying `NomadGasConfig` which is the big struct